### DISCUSSION TITLE
Keyboard navigation of Emanote

### DISCUSSION THREAD

**Discussion by j-steinbach** [srid/emanote#147] [Ideas]
Navigate Emanote with arrow keys (and vim-style `h j k l`), based on the Folgezettel tree.

Left/Right: Cycle between siblings (Notes with the same parent(s)).

Up: Go to parent note. (What happens with multiple parents? Go to first parent? Show an overlay to select the parent?)

Down: Go to child note. (Multiple children: Same as with Up. First child? Select child?)

No parents/siblings/children: Do nothing.

----

See also https://github.com/srid/emanote/discussions/143 (and maybe https://github.com/srid/emanote/discussions/140) for possible alternate display variants that would benefit/support keyboard navigation.

**Comment by vcavallo** [comment]
idea for multiple children/parents:

- press `j` once, see a list of children with one highlighted (maybe at random to prevent over-treading certain trails)
- press `h` and `l` to change which is highlighted
- press `j` again to select it.

Same for `k` with multiple parents.

**Reply by j-steinbach** [reply]
Yeah, this is also what I pictured. The parents/children should probably be shown as a [carousel](https://getbootstrap.com/docs/4.0/components/carousel/) instead of a list to directly "wire-down" the left/right `h`/`l` usage.

**Reply by j-steinbach** [reply]
Quick mockup.

![emanote-hotkey-navigation](https://user-images.githubusercontent.com/69524139/134669313-34a45286-9301-4622-9659-fdc91649fd3c.png)

**Reply by vcavallo** [reply]
:fire: 

How about a preview title of the siblings left and right so you know where you're heading?

**Reply by j-steinbach** [reply]
Absolutely.

![emanote-hotkey-navigation-2](https://user-images.githubusercontent.com/69524139/134908050-34d29d3a-4f38-47ca-ab36-58269b809ba5.png)

**Comment by vcavallo** [comment]
Okay now that I see these mock-ups I _really_ want this :)

On Mon, Sep 27, 2021 at 8:30 AM j-steinbach ***@***.***>
wrote:

> Absolutely.
>
> [image: emanote-hotkey-navigation-2]
> <https://user-images.githubusercontent.com/69524139/134908050-34d29d3a-4f38-47ca-ab36-58269b809ba5.png>
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/srid/emanote/discussions/147#discussioncomment-1390089>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYNQ36FOD24NEBRDMMW57TUEBPWPANCNFSM5EUMVQUQ>
> .
>
-- 
Vinney Cavallo
Founder/Lead Developer, exnil.io

**Comment by j-steinbach** [comment]
Another similar approach is something like [this](https://haxplore.pabue.co/), which might also fit with more https://github.com/srid/emanote/discussions/140
